web/src/buddy_list.js

```diff
@@ -1,16 +1,38 @@
 import $ from "jquery";
+import tippy from "tippy.js";
 
+import render_section_header from "../templates/buddy_list/section_header.hbs";
+import render_view_all_subscribers from "../templates/buddy_list/view_all_subscribers.hbs";
+import render_view_all_users from "../templates/buddy_list/view_all_users.hbs";
+import render_empty_list_widget_for_list from "../templates/empty_list_widget_for_list.hbs";
 import render_presence_row from "../templates/presence_row.hbs";
 import render_presence_rows from "../templates/presence_rows.hbs";
 
 import * as blueslip from "./blueslip";
 import * as buddy_data from "./buddy_data";
+import {media_breakpoints_num} from "./css_variables";
+import * as hash_util from "./hash_util";
+import {$t} from "./i18n";
 import * as message_viewport from "./message_viewport";
+import * as narrow_state from "./narrow_state";
 import * as padded_widget from "./padded_widget";
+import * as peer_data from "./peer_data";
+import * as people from "./people";
 import * as scroll_util from "./scroll_util";
+import * as stream_data from "./stream_data";
+import {INTERACTIVE_HOVER_DELAY} from "./tippyjs";
+
+function get_formatted_sub_count(sub_count) {
+    if (sub_count >= 1000) {
+        // parseInt() is used to floor the value of division to an integer
+        sub_count = Number.parseInt(sub_count / 1000, 10) + "k";
+    }
+    return sub_count;
+}
```

Comment:
This should possibly be using this module:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat

I've not checked `caniuse` etc., or played with it, but the `engineering` style might be the one that spells values this way?

Reply:
Oh yeah this is bad form that's not new here, it's moved from elsewhere in the file. I fixed it when going over typescript. I could just use `Math.floor`, which is what I did there, which feels simplest here and keeps the same logic as was already there.

Reply:
This function is new in this PR, I'm pretty sure. But the main point is internationalization -- I'm not confident every language spells numbers that way.

Reply:
Ohhh I think maybe I was copying this from other code that did something similar then, that's since been deleted. I'm just pretty sure I didn't actually write that code. Sorry for not noticing its jankness when I copied it.

That makes sense about internationalization. I'll look into this!

Reply:
I asked a question about this [here](https://chat.zulip.org/#narrow/stream/101-design/topic/buddy.20list.20split.20users.20.2326717/near/1734019)

Reply:
Updated to use `new Intl.NumberFormat(user_settings.default_language, {notation:  "compact"}).format(sub_count);`